spikeinterface/curation/auto_merge.py

```diff
@@ -0,0 +1,264 @@
+        # ax.axvline(bins2[m + win_size])
+        # dev = -np.gradient(np.gradient(auto_corr))
+        # ax.plot(bins2, dev)
+        # plt.show()
+    
+    
+    
+    
+    
+    corr_diff = np.full((n, n), np.nan, dtype='float64')
+    for unit_ind1 in range(n):
+        for unit_ind2 in range(unit_ind1 + 1, n):
+            if not pair_mask[unit_ind1, unit_ind2]:
+                continue
+
+            unit_id1, unit_id2 = unit_ids[unit_ind1], unit_ids[unit_ind2]
+
+            
+            num1, num2 = num_spikes[unit_id1], num_spikes[unit_id2]
+            # Weighted window (larger unit imposes its window).
+            win_size = int(round((num1 * win_sizes[unit_ind1] + num2 * win_sizes[unit_ind2]) / (num1 + num2)))    
+            # Plage of indices where correlograms are inside the window.
+            corr_inds = np.arange(m - win_size, m + win_size, dtype=int)
+            
+            # TODO : for Aurelien 
+            shift = 0
+
+            auto_corr1 = normalize_correlogram(correlograms_smooth[unit_ind1, unit_ind1, :])
+            auto_corr2 = normalize_correlogram(correlograms_smooth[unit_ind2, unit_ind2, :])
+            cross_corr = normalize_correlogram(correlograms_smooth[unit_ind1, unit_ind2, :])
+            diff1 = np.sum(np.abs(cross_corr[corr_inds - shift] - auto_corr1[corr_inds])) / len(corr_inds)
+            diff2 = np.sum(np.abs(cross_corr[corr_inds - shift] - auto_corr2[corr_inds])) / len(corr_inds)
+            # Weighted difference (larger unit imposes its difference).
+            w_diff = (num1 * diff1 + num2 * diff2) / (num1 + num2)
+            corr_diff[unit_ind1, unit_ind2] = w_diff
+            
+            
+            # debug
+            # corr_thresh = 0.3
+            # if w_diff < corr_thresh:
+            #     import matplotlib.pyplot as plt
+            #     bins2 = bins[:-1] + np.mean(np.diff(bins))
+            #     fig, axs = plt.subplots(ncols=3)
+            #     ax = axs[0]
+            #     ax.plot(bins2, correlograms[unit_ind1, unit_ind1, :], color='b')
+            #     ax.plot(bins2, correlograms[unit_ind2, unit_ind2, :], color='r')
+            #     ax.plot(bins2, correlograms_smooth[unit_ind1, unit_ind1, :], color='b')
+            #     ax.plot(bins2, correlograms_smooth[unit_ind2, unit_ind2, :], color='r')
+                
+                
+            #     ax.set_title(f'{unit_id1}[{num1}] {unit_id2}[{num2}]')
+            #     ax = axs[1]
+            #     ax.plot(bins2, correlograms[unit_ind1, unit_ind2, :], color='g')
+            #     ax = axs[2]
+            #     ax.plot(bins2, auto_corr1, color='b')
+            #     ax.plot(bins2, auto_corr2, color='r')
+            #     ax.axvline(bins2[m - win_size])
+            #     ax.axvline(bins2[m + win_size])
+            #     ax.set_title(f'corr diff {w_diff}')
+            #     plt.show()
+    
+    return corr_diff
+
+def normalize_correlogram(correlogram: np.ndarray):
```

Comment:
`window_ms=50` seems to low for the normalization process (this means computing the correlogram between -25 and +25ms).
I would **at least** put 100, maybe even 150 would be best.

Reply:
OK I put 100.